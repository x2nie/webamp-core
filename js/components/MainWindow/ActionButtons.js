import React from "react";
import { connect } from "react-redux";
import { play, pause, stop } from "../../actionCreators";

const ActionButtons = props => (
  <div className="actions">
    <div id="previous" title="Previous Track" />
    <div id="play" onClick={props.play} title="Play" />
    <div id="pause" onClick={props.pause} title="Pause" />
    <div id="stop" onClick={props.stop} title="Stop" />
    <div id="next" title="Next Track" />
  </div>
);

const mapStateToProps = state => state.media;

const mapDispatchToProps = {
  play,
  pause,
  stop
};

export default connect(mapStateToProps, mapDispatchToProps)(ActionButtons);
