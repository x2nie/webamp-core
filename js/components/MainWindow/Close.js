import React from "react";
import { connect } from "react-redux";

import { close } from "../../actionCreators";

const Close = ({ onClick }) => (
  <div id="close" onClick={onClick} title="Close" />
);

export default connect(null, { onClick: close })(Close);
