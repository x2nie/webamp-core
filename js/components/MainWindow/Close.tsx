import React from "react";
import ClickedDiv from "../ClickedDiv";
import { useActionCreator } from "../../hooks";

import * as Actions from "../../actionCreators";

const Close = React.memo(() => {
  const close = useActionCreator(Actions.close);
  return <ClickedDiv id="close" onClick={close} title="Close" />;
});

export default Close;
