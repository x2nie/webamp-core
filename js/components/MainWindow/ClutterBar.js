import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import {
  SET_FOCUS,
  TOGGLE_DOUBLESIZE_MODE,
  UNSET_FOCUS
} from "../../actionTypes";

const ClutterBar = props => (
  <div id="clutter-bar">
    <div id="button-o" />
    <div id="button-a" />
    <div id="button-i" />
    <div
      title={"Toggle Doublesize Mode"}
      id="button-d"
      className={classnames({ selected: props.doubled })}
      onMouseUp={props.handleMouseUp}
      onMouseDown={props.handleMouseDown}
    />
    <div id="button-v" />
  </div>
);

const mapStateToProps = state => ({
  doubled: state.display.doubled
});

const mapDispatchToProps = dispatch => ({
  handleMouseDown: () => dispatch({ type: SET_FOCUS, input: "double" }),
  handleMouseUp: () => {
    dispatch({ type: TOGGLE_DOUBLESIZE_MODE });
    dispatch({ type: UNSET_FOCUS });
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ClutterBar);
