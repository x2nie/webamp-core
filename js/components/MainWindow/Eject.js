import React from "react";
import { connect } from "react-redux";

import { openFileDialog } from "../../actionCreators";

const Eject = props => (
  <div id="eject" onClick={props.openFileDialog} title="Open File(s)" />
);

const mapDispatchToProps = (dispatch, ownProps) => ({
  openFileDialog: () => dispatch(openFileDialog(ownProps.fileInput))
});

export default connect(() => ({}), mapDispatchToProps)(Eject);
