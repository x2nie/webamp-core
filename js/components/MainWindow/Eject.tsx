import React from "react";

import * as Actions from "../../actionCreators";
import { useActionCreator } from "../../hooks";

const Eject = React.memo(() => {
  const openMediaFileDialog = useActionCreator(Actions.openMediaFileDialog);
  return <div id="eject" onClick={openMediaFileDialog} title="Open File(s)" />;
});

export default Eject;
