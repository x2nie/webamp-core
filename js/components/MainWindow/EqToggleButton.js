import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { getWindowOpen } from "../../selectors";
import { toggleWindow } from "../../actionCreators";

const EqToggleButton = props => (
  <div
    id="equalizer-button"
    className={classnames({ selected: props.getWindowOpen("equalizer") })}
    onClick={props.handleClick}
    title="Toggle Graphical Equalizer"
  />
);

const mapStateToProps = state => ({
  getWindowOpen: getWindowOpen(state)
});

const mapDispatchToProps = {
  handleClick: () => toggleWindow("equalizer")
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EqToggleButton);
