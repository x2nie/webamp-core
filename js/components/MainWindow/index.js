import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { WINDOWS } from "../../constants";

import MiniTime from "../MiniTime";

import ActionButtons from "./ActionButtons";
import MainBalance from "./MainBalance";
import Close from "./Close";
import ClutterBar from "./ClutterBar";
import MainContextMenu from "./MainContextMenu";
import Eject from "./Eject";
import EqToggleButton from "./EqToggleButton";
import PlaylistToggleButton from "./PlaylistToggleButton";
import Kbps from "./Kbps";
import Khz from "./Khz";
import Marquee from "./Marquee";
import MonoStereo from "./MonoStereo";
import Position from "./Position";
import Repeat from "./Repeat";
import Shade from "./Shade";
import Shuffle from "./Shuffle";
import Time from "./Time";
import Visualizer from "./Visualizer";
import MainVolume from "./MainVolume";

import { SET_FOCUSED_WINDOW } from "../../actionTypes";

import { loadFileFromReference } from "../../actionCreators";

import "../../../css/main-window.css";

export class MainWindow extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.handleDrop = this.handleDrop.bind(this);
  }

  handleClick() {
    this.props.setFocus();
  }

  supress(e) {
    e.stopPropagation();
    e.preventDefault();
  }

  handleDrop(e) {
    this.supress(e);
    const { files } = e.dataTransfer;
    // TODO: Move this to an actionCreator
    this.props.dispatch(loadFileFromReference(files[0]));
  }

  render() {
    const {
      focused,
      loading,
      doubled,
      shade,
      llama,
      status,
      working
    } = this.props;

    const className = classnames({
      window: true,
      // TODO: Handle these status changes in the individual components
      play: status === "PLAYING",
      stop: status === "STOPPED",
      pause: status === "PAUSED",
      selected: focused === WINDOWS.MAIN,
      draggable: true,
      loading,
      doubled,
      llama,
      shade
    });

    return (
      <div
        id="main-window"
        className={className}
        onMouseDown={this.handleClick}
        onDragEnter={this.supress}
        onDragOver={this.supress}
        onDrop={this.handleDrop}
      >
        <div id="title-bar" className="selected title-bard draggable">
          <MainContextMenu fileInput={this.props.fileInput} />
          {shade && <MiniTime />}
          <div id="minimize" />
          <Shade />
          <Close />
        </div>
        <div className="status">
          <ClutterBar />
          <div id="play-pause" />
          <div
            id="work-indicator"
            className={classnames({ selected: working })}
          />
          <Time />
          <Visualizer analyser={this.props.mediaPlayer._analyser} />
        </div>
        <div className="media-info">
          <Marquee />
          <Kbps />
          <Khz />
          <MonoStereo />
        </div>
        <MainVolume />
        <MainBalance />
        <div className="windows">
          <EqToggleButton />
          <PlaylistToggleButton />
        </div>
        <Position />
        <ActionButtons />
        <Eject fileInput={this.props.fileInput} />
        <div className="shuffle-repeat">
          <Shuffle />
          <Repeat />
        </div>
        <a
          id="about"
          target="blank"
          href="https://github.com/captbaritone/winamp2-js"
          title="About"
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const {
    media: { status },
    display: { loading, doubled, shade, llama, working },
    windows: { focused }
  } = state;
  return { status, loading, doubled, shade, llama, working, focused };
};

const mapDispatchToProps = dispatch => ({
  setFocus: () => {
    dispatch({ type: SET_FOCUSED_WINDOW, window: WINDOWS.MAIN });
  }
});
export default connect(mapStateToProps, mapDispatchToProps)(MainWindow);
