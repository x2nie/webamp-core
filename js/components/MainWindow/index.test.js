const Winamp = {
  media: {
    addEventListener: jest.fn(),
    _analyser: null
  }
};

const canvasMockify = require("canvas-mock");

function createNodeMock(element) {
  if (element.type === "canvas") {
    const Canvas = window.document.createElement("canvas");
    canvasMockify(Canvas); // mock canvas functions required by Phaser.js are added
    return Canvas;
  }
  return null;
}

window.requestAnimationFrame = () => {};

import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import getStore from "../../store";

import MainWindow from "./index";

describe("MainWindow", () => {
  let store;
  beforeEach(() => {
    store = getStore(Winamp);
  });

  it("renders to snapshot", () => {
    const options = { createNodeMock };
    const tree = renderer
      .create(
        <Provider store={store}>
          <MainWindow winamp={Winamp} mediaPlayer={Winamp.media} />
        </Provider>,
        options
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
