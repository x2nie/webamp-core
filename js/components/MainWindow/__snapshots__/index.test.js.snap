// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWindow renders to snapshot 1`] = `
<div
  className="window stop selected draggable loading"
  id="main-window"
  onDragEnter={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="selected title-bard draggable"
    id="title-bar"
  >
    <div
      id="option-context"
    >
      <div
        className="handle"
        onClick={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          id="option"
          title="Winamp Menu"
        />
      </div>
    </div>
    <div
      id="minimize"
    />
    <div
      id="shade"
      onClick={[Function]}
      title="Toggle Windowshade Mode"
    />
    <div
      id="close"
      onClick={[Function]}
      title="Close"
    />
  </div>
  <div
    className="status"
  >
    <div
      id="clutter-bar"
    >
      <div
        id="button-o"
      />
      <div
        id="button-a"
      />
      <div
        id="button-i"
      />
      <div
        className=""
        id="button-d"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        title="Toggle Doublesize Mode"
      />
      <div
        id="button-v"
      />
    </div>
    <div
      id="play-pause"
    />
    <div
      className=""
      id="work-indicator"
    />
    <div
      className="countdown"
      id="time"
      onClick={[Function]}
    >
      <div
        className="digit digit-0"
        id="minute-first-digit"
      />
      <div
        className="digit digit-0"
        id="minute-second-digit"
      />
      <div
        className="digit digit-0"
        id="second-first-digit"
      />
      <div
        className="digit digit-0"
        id="second-second-digit"
      />
    </div>
    <canvas
      height="32"
      id="visualizer"
      onClick={[Function]}
      width="152"
    />
  </div>
  <div
    className="media-info"
  >
    <div
      className="text"
      id="marquee"
      onMouseDown={[Function]}
      title="Song Title"
    >
      <div
        style={
          Object {
            "marginLeft": "0px",
          }
        }
      >
        <div
          className=" character character-119"
        >
          W
        </div>
        <div
          className=" character character-105"
        >
          i
        </div>
        <div
          className=" character character-110"
        >
          n
        </div>
        <div
          className=" character character-97"
        >
          a
        </div>
        <div
          className=" character character-109"
        >
          m
        </div>
        <div
          className=" character character-112"
        >
          p
        </div>
        <div
          className=" character character-32"
        >
           
        </div>
        <div
          className=" character character-50"
        >
          2
        </div>
        <div
          className=" character character-46"
        >
          .
        </div>
        <div
          className=" character character-57"
        >
          9
        </div>
        <div
          className=" character character-49"
        >
          1
        </div>
        ;
      </div>
    </div>
    <div
      id="kbps"
    >
      <div
        className=" character character-110"
      >
        n
      </div>
      <div
        className=" character character-117"
      >
        u
      </div>
      <div
        className=" character character-108"
      >
        l
      </div>
      <div
        className=" character character-108"
      >
        l
      </div>
      ;
    </div>
    <div
      id="khz"
    >
      <div
        className=" character character-110"
      >
        n
      </div>
      <div
        className=" character character-117"
      >
        u
      </div>
      <div
        className=" character character-108"
      >
        l
      </div>
      <div
        className=" character character-108"
      >
        l
      </div>
      ;
    </div>
    <div
      className="mono-stereo"
    >
      <div
        className=""
        id="stereo"
      />
      <div
        className=""
        id="mono"
      />
    </div>
  </div>
  <input
    className={undefined}
    id="volume"
    max="100"
    min="0"
    onChange={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    step="1"
    style={
      Object {
        "backgroundPosition": "0 -195px",
      }
    }
    title="Volume Bar"
    type="range"
    value={50}
  />
  <input
    className={undefined}
    id="balance"
    max="100"
    min="-100"
    onChange={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    step="1"
    style={
      Object {
        "backgroundPosition": "0 --15px",
      }
    }
    title="Panning Bar"
    type="range"
    value={0}
  />
  <div
    className="windows"
  >
    <div
      className="selected"
      id="equalizer-button"
      onClick={[Function]}
      title="Toggle Graphical Equalizer"
    />
    <div
      className=""
      id="playlist-button"
      onClick={[Function]}
      title="Toggle Playlist Editor"
    />
  </div>
  <input
    className="left"
    id="position"
    max="100"
    min="0"
    onChange={[Function]}
    onInput={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    step="1"
    title="Seeking Bar"
    type="range"
    value={0}
  />
  <div
    className="actions"
  >
    <div
      id="previous"
      title="Previous Track"
    />
    <div
      id="play"
      onClick={[Function]}
      title="Play"
    />
    <div
      id="pause"
      onClick={[Function]}
      title="Pause"
    />
    <div
      id="stop"
      onClick={[Function]}
      title="Stop"
    />
    <div
      id="next"
      title="Next Track"
    />
  </div>
  <div
    id="eject"
    onClick={[Function]}
    title="Open File(s)"
  />
  <div
    className="shuffle-repeat"
  >
    <div
      className=""
      id="shuffle"
      onClick={[Function]}
      title="Toggle Shuffle"
    />
    <div
      className=""
      id="repeat"
      onClick={[Function]}
      title="Toggle Repeat"
    />
  </div>
  <a
    href="https://github.com/captbaritone/winamp2-js"
    id="about"
    target="blank"
    title="About"
  />
</div>
`;
